workspaces/cli-shared/src/config.ts

```diff
@@ -0,0 +1,72 @@
+import semver from 'semver';
+import { PackageJson } from 'type-fest';
+
+export { PackageJson };
+
+export interface CliConfigObject {
+  name: string;
+  version: string;
+  channel: string;
+  prerelease: boolean;
+
+  envName: Env;
+  env: {
+    development: boolean;
+    production: boolean;
+  };
+}
+
+export enum Env {
+  Development = 'development',
+  Staging = 'staging',
+  Testing = 'testing',
+  Production = 'production',
+}
+
+export function create(
+  pkg: PackageJson,
+  env: { [key: string]: string | undefined }
+): CliConfigObject {
+  const name = pkg.name;
+  if (!name) {
+    throw new Error(
+      'Could not create cli config object: package.json must have a name'
+    );
+  }
+  const version = semver.valid(pkg.version);
+  if (!version) {
+    throw new Error(
+      `Could not create cli config object for ${name}: version in package.json not valid semver`
+    );
+  }
+  const prereleaseComponents = semver.prerelease(version);
+  const prereleaseTag =
+    prereleaseComponents &&
+    prereleaseComponents.find(
+      (component: string | number) => typeof component === 'string'
+    );
+
+  const isPrerelease = !!prereleaseComponents;
+  const channel = isPrerelease ? prereleaseTag || 'prerelease' : 'latest';
+
+  const envName = isEnvTrue(env.OPTIC_DEVELOPMENT)
+    ? Env.Development
+    : Env.Production;
+
+  return {
+    name,
+    version,
+    channel,
+    prerelease: isPrerelease,
+
+    envName,
+    env: {
+      development: envName === Env.Development,
+      production: envName === Env.Production,
+    },
+  };
+}
+
+export function isEnvTrue(val: string | undefined) {
+  return val === 'yes' || val === 'true' || val === '1' || val === 'y';
```

Comment:
As an aside I'd strongly prefer we did not use anything except yes/no

Reply:
*_ENABLED=yes|no|undefined/empty, and checking if it is yes means opt-in and checking if no means opt-out

Reply:
we're already using `true` for feature flags as well, which is the main reason why I added this. Added `1` and `y` as that's super common in other Unixy / GNU tools, but happy to leave those out since they aren't used anywhere.